lib/combine_pdf/parser.rb

```diff
@@ -335,7 +335,7 @@ def _parse_
               str << str_bytes.shift
             end
           end
-          out << unify_string(str.pack('C*').force_encoding(Encoding::ASCII_8BIT))
+          out << unify_string(str.pack('C*').dup.force_encoding(Encoding::ASCII_8BIT))
```

Comment:
(again) This dup is redundant and will use up a bunch of memory. The dynamic String returned from `Array#pack` is mutable.